money.py

```diff
@@ -0,0 +1,44 @@
+"""The module provides Money class
+"""
+from currency import Currency, CURRENCY_CACHE
+
+
+class Money:
+    """Class represents money, as well as some maths
+    """
+    value = Currency()
+
+    def __init__(self, value, currency="USD"):
+        self.currency = currency
+        self.value = value
+
+    def __str__(self):
+        return "<{0}: {1} {2}>".format(self.__class__.__name__,
+                                       round(self.value, 2),
+                                       self.currency)
+
+    def __add__(self, other):
+        rate = CURRENCY_CACHE[f"{other.currency}_{self.currency}"]
+
+        return self.__class__(self.value + other.value * rate, self.currency)
```

Comment:
it would be better to use `Money(...)` instead of `self.__class__` as according to PEP20: Explicit is better than implicit. Readability counts.
However, there is a sense to use such construction for mixin class.

Reply:
But `Beautiful is better than ugly.` ;)